### DISCUSSION TITLE
Tutorial 4: Verify that your monitoring stack has been correctly configured

### DISCUSSION THREAD

**Discussion by nyameko** [chpc-tech-eval/scc#158] [Show and tell]
Post screenshots of your outputs similar to the following:

```bash
$ sudo docker compose up -d
[+] Running 4/4
✔ Network monitoring_stack_monitoring-network Created 0.3s
✔ Container monitoring_stack-grafana-1 Started 0.2s
✔ Container monitoring_stack-prometheus-1 Started 0.3s
✔ Container monitoring_stack-node-exporter-1 Started 0.2s
```
And

```bash
sudo docker ps
CONTAINER ID   IMAGE                COMMAND                  CREATED      STATUS        PORTS                                       NAMES
2b707570dc41   grafana/grafana      "/run.sh"                6 days ago   Up 12 hours   0.0.0.0:3000->3000/tcp, :::3000->3000/tcp   monitoring_stack-grafana-1
ea730ef94381   prom/prometheus      "/bin/prometheus --c…"   6 days ago   Up 10 hours   0.0.0.0:9090->9090/tcp, :::9090->9090/tcp   monitoring_stack-prometheus-1
704dfa94ecf3   prom/node-exporter   "/bin/node_exporter"     6 days ago   Up 12 hours   0.0.0.0:9100->9100/tcp, :::9100->9100/tcp   monitoring_stack-node-exporter-1
```

Dump database output from Prometheus, Node-Exporter and Grafana:
```bash
$ curl -s localhost:9090/metrics | head
# HELP go_gc_cycles_automatic_gc_cycles_total Count of completed GC cycles generated by the Go runtime.
# TYPE go_gc_cycles_automatic_gc_cycles_total counter
go_gc_cycles_automatic_gc_cycles_total 5
# HELP go_gc_cycles_forced_gc_cycles_total Count of completed GC cycles forced by the application.
# TYPE go_gc_cycles_forced_gc_cycles_total counter
go_gc_cycles_forced_gc_cycles_total 0
# HELP go_gc_cycles_total_gc_cycles_total Count of all completed GC cycles.
# TYPE go_gc_cycles_total_gc_cycles_total counter
go_gc_cycles_total_gc_cycles_total 5
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 7.81e-05
go_gc_duration_seconds{quantile="0.25"} 0.000135
...

$ curl -s localhost:9100/metrics | head
# HELP go_gc_duration_seconds A summary of the pause duration of
garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 2.1937e-05
go_gc_duration_seconds{quantile="0.25"} 3.2322e-05
go_gc_duration_seconds{quantile="0.5"} 3.4946e-05
go_gc_duration_seconds{quantile="0.75"} 5.7424e-05
go_gc_duration_seconds{quantile="1"} 0.000171199
go_gc_duration_seconds_sum 0.007451006
go_gc_duration_seconds_count 157
# HELP go_goroutines Number of goroutines that currently exist

$ curl -s localhost:3000 | hea
<a href=“/login">Found</a>.
$ curl -s -u admin:admin localhost:3000 | head
<!DOCTYPE html>
<html lang="en-US">
<head>
<meta charset="utf-8" />
<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
<meta name="viewport" content="width=device-width" />
<meta name="theme-color" content="#000" />
<title>Grafana</title>
```

**Comment by CraftierLamb599** [comment]
# Wits-1
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/137203651/c83e07e0-fdce-445f-ad11-c9191fec8d18)
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/137203651/982fa11b-049c-4e6c-b71c-d5fd76af732d)
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/137203651/ef861727-c3cb-4d83-bd04-a0cdb9d631be)
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/137203651/84696692-83c3-4922-83b2-a64d0f401e91)

**Comment by Kohzmik** [comment]
CUT_Team
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/138365090/4253a491-0894-4018-96c6-0a62a1a6c241)
AND
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/138365090/fda8d7ad-a8cc-4a43-a3f3-dde0fc61b067)
AND
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/138365090/7bc1a4a5-8be3-43e8-a5da-e979efe32e8c)

**Comment by Dc-Spears** [comment]
**CPUT!** - DC Spears

 
![1](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175037009/39a5c013-668c-4b5d-8834-04f4f5e81f7c)
![2](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175037009/24858489-7505-407b-bbd2-342b755ea389)
![3](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175037009/94d739a2-fe3c-461c-b3ad-d6ec291024e7)

**Comment by lisabeeaan** [comment]
**HPC Wizards**
![WhatsApp Image 2024-07-11 at 14 01 43_1b1571ee](https://github.com/user-attachments/assets/e63bc7ed-493d-4203-bdfd-ffa779ceb653)
![WhatsApp Image 2024-07-11 at 14 04 27_2b100637](https://github.com/user-attachments/assets/76521ad8-7ad1-48aa-8e3a-6e6434e5bc0d)
AND dumping database output from Prometheus, Node-Exporter and Grafana
![image](https://github.com/user-attachments/assets/c5d55ceb-0508-4bce-86c5-2e139469a84e)

**Comment by ThabangMmusi** [comment]
### Parallel Pioneers
![Screenshot 2024-07-12 005908](https://github.com/user-attachments/assets/9d91143f-7ef4-49ae-ae72-5b34caac9d2a)
![Screenshot 2024-07-12 005934](https://github.com/user-attachments/assets/17d183d9-2c17-424e-b318-90d089f227aa)
![Screenshot 2024-07-12 013614](https://github.com/user-attachments/assets/9caa1370-2a1b-4d47-b2da-950a39259f3d)

**Comment by Talhahpatelia** [comment]
## Wits-A

1. Bring up your monitoring stack

<img width="475" alt="stack" src="https://github.com/user-attachments/assets/8877136f-a901-4a7f-bbe3-cc64a1d98214">

2. Dump the metrics that are being monitored from your services

<img width="742" alt="dump" src="https://github.com/user-attachments/assets/fc679346-bfa8-45eb-9937-ef6fc9b8884e">

3. Logged into Grafana

<img width="551" alt="logged" src="https://github.com/user-attachments/assets/5c0ea32a-34ee-483c-a324-30899c2f78b9">

**Comment by lisabeeaan** [comment]
**HPC Wizards**
 The database output from Prometheus, Node-Exporter and Grafana as well as Grafana login.
![image](https://github.com/user-attachments/assets/1624f96a-72cc-4e98-9755-6d4ae23c1ed5)

**Comment by ADELAIDEMOGALE** [comment]
### **Khulisa Data**

![Screenshot (711)](https://github.com/user-attachments/assets/96eaa390-ae06-48b9-997b-1c80d491cc2f)

![Screenshot ](https://github.com/user-attachments/assets/20f70d1a-e31e-4034-9bdd-2e29baf3672c)

![Screenshot (710)](https://github.com/user-attachments/assets/aff16903-3391-4293-aac4-d2a2498c51cc)

**Comment by 15kay** [comment]
HYPERNET
![image](https://github.com/user-attachments/assets/d9c51c96-14ea-475b-b5a6-82718936e8f5)

**Comment by Rosemary2301** [comment]
Algorithm Avengers

sudo docker compose up -d
![sudo docker compose up -d](https://github.com/user-attachments/assets/4d6c4ffd-63aa-439f-aea2-cc28b5edb7c6)

sudo docker ps
![sudo docker ps](https://github.com/user-attachments/assets/a47eed2b-ac65-4414-a87f-0e85a4f1794c)

**Comment by SfisoMncube** [comment]
**Khulisa IP-A**

![image](https://github.com/user-attachments/assets/6340a0a3-759d-4020-85fa-a85f7c7a5eb0)

**_Dump database output from Prometheus, Node-Exporter and Grafana_**

![image](https://github.com/user-attachments/assets/7ab550a4-01ee-44f3-9c04-c34d98d0eee0)

**Comment by WilliamMohlahlane** [comment]
### **ThunderBytes**
![WhatsApp Image 2024-07-13 at 06 58 08](https://github.com/user-attachments/assets/3ae469ee-6ffa-437c-958c-287ee35fbbce)

**Comment by ThabangMmusi** [comment]
### Parallel Pioneers
**Successfully logged in to JupyterLab**

![Screenshot 2024-07-13 075955](https://github.com/user-attachments/assets/32dcfcc6-1152-4589-b802-99e607e0ac0c)
![Screenshot 2024-07-13 080524](https://github.com/user-attachments/assets/39efcad6-0804-48df-8099-627876fd55f1)
![Screenshot 2024-07-13 080533](https://github.com/user-attachments/assets/61cf0a6a-2200-4827-8ba1-ac5418d35802)
![Screenshot 2024-07-13 080548](https://github.com/user-attachments/assets/eb0f11b9-4114-4dc1-930f-43f61303499a)
![Screenshot 2024-07-13 080626](https://github.com/user-attachments/assets/a6ae8364-de4b-4d0f-8297-50552936e6bb)

**Comment by AnnaLukose** [comment]
EEElite (UCT)
![1](https://github.com/user-attachments/assets/8e5cc39a-98d3-436f-9f2a-b2e1799310e7)
![2](https://github.com/user-attachments/assets/8a80c972-b034-4372-b376-ad48fbc7acb6)

**Comment by Cyphers-UFH** [comment]
uFH -Cyphers

![Screenshot 2024-07-13 094242](https://github.com/user-attachments/assets/e65e8b7e-dcee-48b5-8805-f0568641ee5e)
![Screenshot 2024-07-13 094711](https://github.com/user-attachments/assets/c0a8965a-80bb-466a-9658-639fbf85946b)

**Comment by huziibee** [comment]
**Wits-i**
![image](https://github.com/user-attachments/assets/c3e5b6bf-55f9-4354-a94a-43bece81473d)

**Comment by Aatikah-Sheikh** [comment]
Wits-A


![Screenshot 2025-07-10 120922](https://github.com/user-attachments/assets/d2913842-4fbd-4516-8ab3-7648aa21c573)

Prometheus (9090):
Shows Prometheus’ own metrics like Go runtime stats.
Confirms Prometheus is running and exposing data.

Node Exporter (9100):
Outputs system metrics for the host machine.
Proves Node Exporter is active and ready to be scraped.

Grafana (3000):
Returns HTML for the login page.
Verifies Grafana’s web UI is up and reachable.

**Comment by MphephuSamuel** [comment]
computer champions

<img width="1919" height="1079" alt="Screenshot 2025-07-10 001809" src="https://github.com/user-attachments/assets/62e83b02-a656-4d59-b723-24f5517deec1" />

**Comment by makmak-alt** [comment]
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/488a8eab-e60f-4cb9-9309-f10aea1be5c3" />

Ctrl+Alt+Defeat

**Comment by Cloud-Strive** [comment]
Runtime Blinders

<img width="1366" height="768" alt="docker dumping" src="https://github.com/user-attachments/assets/345588ae-48a8-46c7-a6d8-803328037878" />

**Comment by ShavirV** [comment]
TuksByters
<img width="1896" height="716" alt="image" src="https://github.com/user-attachments/assets/54d1de3d-1174-42a1-b9d0-3566d28ad29c" />

**Comment by mpolokeng8** [comment]
DebugThugs

<img width="1920" height="1080" alt="Screenshot (269)" src="https://github.com/user-attachments/assets/86636ab0-074a-4865-bbaa-4ea8735b4136" />

**Comment by KayGee-17** [comment]
CybertronNextGen

Docker compose up
<img width="1365" height="753" alt="compose d" src="https://github.com/user-attachments/assets/eff73ffa-ab12-4710-965d-40ab399cc9f1" />

Dataset Dump
<img width="1362" height="655" alt="database dump" src="https://github.com/user-attachments/assets/e68ba083-3ff6-4a99-93ff-02475dbb4b76" />

Docker ps
<img width="1358" height="151" alt="docker ps" src="https://github.com/user-attachments/assets/918e15ff-4c6e-4c68-8f87-e08fe9eb2f34" />

**Comment by MahlogonoloR** [comment]
DC DAGGERS

<img width="1920" height="1080" alt="Screenshot 2025-07-11 041232" src="https://github.com/user-attachments/assets/832b39d3-0b03-429b-9db2-35bf9508419a" />

**Comment by Iloke-Alusala** [comment]
0V1RCL0CK1D
<img width="1179" height="739" alt="Screenshot 2025-07-11 at 14 26 59" src="https://github.com/user-attachments/assets/14f7be99-a8ea-4781-a289-576737e51208" />

**Comment by ethannef** [comment]
Wits1

<img width="1893" height="123" alt="image" src="https://github.com/user-attachments/assets/71e05ec7-83d6-4d0e-8a26-78b93953c9c0" />

<img width="1916" height="103" alt="image" src="https://github.com/user-attachments/assets/cf41c397-67bb-4b19-a519-75a185f81a21" />

<img width="1919" height="683" alt="image" src="https://github.com/user-attachments/assets/ab1428a4-53b7-45fc-bd98-efffd8c9c690" />

**Comment by 111nation** [comment]
**UPtime**

<img width="1841" height="963" alt="image" src="https://github.com/user-attachments/assets/225ae98a-97dc-4da3-862d-6e531dad9afc" />

**Comment by Valencia-byte** [comment]
<img width="1265" height="381" alt="Screenshot 2026-07-08 213646" src="https://github.com/user-attachments/assets/b8ee4b99-3c08-4104-b1dd-7e1ba47fb050" />
<img width="1260" height="136" alt="Screenshot 2026-07-08 213659" src="https://github.com/user-attachments/assets/14275fdf-19ce-48ee-8422-e67e22b75c92" />
<img width="1254" height="494" alt="Screenshot 2026-07-08 213626" src="https://github.com/user-attachments/assets/a800fded-71e3-4ffe-87c2-4dec104420fd" />

G3MIKNIGHTS Docker containers running

**Comment by ntando-memela** [comment]
WITS!-2026
<img width="1851" height="253" alt="image" src="https://github.com/user-attachments/assets/785c045b-2658-4418-b1e6-12c3b6c5bfe8" />
<img width="1852" height="614" alt="Screenshot from 2026-07-08 22-07-06" src="https://github.com/user-attachments/assets/c919752c-9e74-4658-af1e-46e7a32acbb8" />

**Comment by switchsquad** [comment]
<img width="1912" height="442" alt="Screenshot 2026-07-09 141414" src="https://github.com/user-attachments/assets/314d2b16-0334-46ed-bdaa-910beb194414" />
<img width="1897" height="926" alt="Screenshot 2026-07-09 141805" src="https://github.com/user-attachments/assets/94de6144-67f7-476d-864a-3256401acb6b" />

Switch Squad(TUT)

**Comment by mabotenick97-cloud** [comment]
CyberOps 
<img width="1858" height="1004" alt="Screenshot from 2026-07-09 15-37-15" src="https://github.com/user-attachments/assets/4487d376-9657-45f8-950c-454a9f4ee9b6" />

**Comment by CadmusKei** [comment]
Jargon (UWC) | SCC 2026 | Verify monitor stack has been correctly configured 
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/22e0c18f-59a3-4b43-b8bc-df0bb463a3a9" />

**Comment by JayJay-lab** [comment]
Sol_Plaatje_A
<img width="1920" height="1200" alt="Screenshot (399)" src="https://github.com/user-attachments/assets/da238201-25a2-431a-813e-10f0514bcf48" />        
<img width="1920" height="1200" alt="Screenshot (400)" src="https://github.com/user-attachments/assets/53905996-c084-4c5f-9dd1-053225f9e2ad" />

**Comment by Slindokuhle19** [comment]
UnizuluComputeX
<img width="1895" height="972" alt="Screenshot 2026-07-09 232149" src="https://github.com/user-attachments/assets/5e41f12a-3424-4ccf-8b78-215571ff58fd" />

**Comment by ThapeloVundla** [comment]
DC Spears

<img width="1908" height="1015" alt="image" src="https://github.com/user-attachments/assets/9ef11678-6eeb-46f2-88e4-92bbc9be3211" />

**Comment by saxs-14** [comment]
3ncrypt3d (UMP)
<img width="1867" height="451" alt="Screenshot 2026-07-10 021013" src="https://github.com/user-attachments/assets/2316da09-58c9-4241-8767-c4ea6eacd3c5" />
<img width="1887" height="672" alt="Screenshot 2026-07-10 021123" src="https://github.com/user-attachments/assets/78e4908f-2135-4dce-b169-8b94d34542ac" />

**Comment by majaha8** [comment]
NEBULA
<img width="1857" height="1014" alt="Screenshot from 2026-07-10 08-04-58" src="https://github.com/user-attachments/assets/5d8a1884-4dd2-459c-84c2-6d6c26d8d43f" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
Using Docker to bring  the monitoring stack
<img width="1622" height="224" alt="image" src="https://github.com/user-attachments/assets/701032b3-92cc-4953-97de-43e128efda34" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
Confirming the status of the Docker containers
<img width="1768" height="142" alt="image" src="https://github.com/user-attachments/assets/bd841bda-cddc-4cef-85fe-07a5aeba041f" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
Docker monitored metrics(Prometheus)
<img width="1861" height="301" alt="image" src="https://github.com/user-attachments/assets/aaf71b4e-48ac-4ca5-a6bd-27564502fb8c" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
Docker monitored metrics(Node Exporter)
<img width="1848" height="272" alt="image" src="https://github.com/user-attachments/assets/80d4c0cb-d4e7-44e3-8326-9266505a8e60" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
Docker monitored metrics(Grafana)
<img width="879" height="71" alt="image" src="https://github.com/user-attachments/assets/0bd88188-1284-4f4c-bdbe-69d562c90e08" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
Docker monitoring metices for Grafana,Node Exporter and Prometheus on the ComputeNode1
<img width="1865" height="795" alt="image" src="https://github.com/user-attachments/assets/4ea9f6a3-0fcc-415a-8cee-233a8ad57c11" />

**Comment by Cleve0602** [comment]
### DC Axes (CPUT) | SCC 2026 ### 
Verification that monitor stack has completely be configured
<img width="1920" height="1080" alt="OuputOf3MonitoringServices" src="https://github.com/user-attachments/assets/564e6942-5f01-47a4-9cab-06d2b7f511f7" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
Docker monitoring metices for Grafana,Node Exporter and Prometheus on the ComputeNode2

<img width="1863" height="963" alt="image" src="https://github.com/user-attachments/assets/a7744b35-a9ed-4c81-a689-10ba5fa73765" />

**Comment by Lihle0512** [comment]
ChaosCompilers

<img width="990" height="955" alt="Screenshot from 2026-07-10 08-18-54" src="https://github.com/user-attachments/assets/0ef5b832-5337-42ee-8e64-5306c543266f" />
<img width="990" height="955" alt="Screenshot from 2026-07-10 08-26-44" src="https://github.com/user-attachments/assets/6c0b3371-1a42-4e6d-a2fa-8fbe3e6e623d" />
